### PR TITLE
Feat: Input refactoring

### LINKED ISSUE #2587: UI. Update Input component
- Input should not accept `className` & `wrapperClass`
- Move styles from wrapper component to native `<input />`
- Adjust prefix/suffix elements in order to input have proper padding
- Fix `SearchInput` clear button (doesn't work currently)

related to #2490

### PR DESCRIPTION
**Pull request by tuul-wq**
- Updated `Input` content
- Moved Inputs to `ui-kit`
- Updated related imports and screens

closes #2587 

Before:
image

After:
image

### CODE REVIEW

File: src/renderer/app/index.css [hunk 1]
```diff
@@ -69,3 +69,16 @@ ul:focus-visible {
 .scrollbar-stable {
   scrollbar-gutter: stable both-edges;
 }
+
+.visually-hidden {
```
Comment (tuul-wq) [inline src/renderer/app/index.css:79]: Well known styles that "hide" element, really useful with label + input (as instance: type="file")

File: src/renderer/features/contacts/ContactFilter/model/contact-filter.ts [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 
 import { includes } from '@/shared/lib/utils';
 import { contactModel } from '@/entities/contact';
 
 const formInitiated = createEvent();
-
-const $filterQuery = createStore<string>('');
 const queryChanged = createEvent<string>();
 
+const $filterQuery = restore(queryChanged, '');
```
Comment (Asmadek) [inline src/renderer/features/contacts/ContactFilter/model/contact-filter.ts]: Maybe simplify it and call just `query`?

File: src/renderer/features/wallets/WalletSelect/ui/WalletPanel.tsx
Comment (Asmadek) [inline src/renderer/features/wallets/WalletSelect/ui/WalletPanel.tsx:1]: Empty file?

File: src/renderer/features/balances/components/AmountInput.tsx [hunk 3]
```diff
@@ -196,3 +197,71 @@ export const AmountInput = ({
...
+  value: string;
+  placeholder: string;
+  label: ReactNode;
+  invalid?: boolean;
+  disabled?: boolean;
+  prefixElement: ReactNode;
+  suffixElement?: ReactNode;
+  onChange: (value: string) => void;
+};
+export const Input = ({ label, invalid, disabled, prefixElement, suffixElement, onChange, ...props }: InputProps) => {
```
Comment (Asmadek) [inline src/renderer/features/balances/components/AmountInput.tsx]: Maybe split it for two files?

**Review by Asmadek — APPROVED** [review]

File: src/renderer/features/balances/index.ts
Comment (johnthecat) [inline src/renderer/features/balances/index.ts:1]: Why this feature called balances? Can you be more specific about what it's about?

File: src/renderer/shared/ui-kit/Inputs/Input/Input.test.tsx
Comment (johnthecat) [inline src/renderer/shared/ui-kit/Inputs/Input/Input.test.tsx:1]: Move this test inside storybook play function

File: src/renderer/shared/ui-kit/Inputs/Input/Input.stories.tsx [hunk 4]
```diff
@@ -42,9 +42,16 @@ Disabled.args = {
   disabled: true,
 };
 
+export const Prefix = Template.bind({});
```
Comment (johnthecat) [inline src/renderer/shared/ui-kit/Inputs/Input/Input.stories.tsx]: Use latest story format, please. You can see examples in the other ui-kit components

File: src/renderer/shared/ui-kit/Inputs/Input/Input.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+      label,
+      invalid,
+      disabled,
+      autoFocus,
+      spellCheck = false,
+      prefixElement,
+      suffixElement,
+      onChange,
+      onPaste,
+      ...props
```
Comment (johnthecat) [inline src/renderer/shared/ui-kit/Inputs/Input/Input.tsx]: Why?

File: src/renderer/shared/ui-kit/Inputs/common/types.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-type BaseHTMLInputProps = 'value' | 'required' | 'disabled' | 'placeholder' | 'name' | 'className' | 'autoFocus';
+type BaseHTMLInputProps = 'value' | 'required' | 'disabled' | 'placeholder' | 'name' | 'autoFocus';
```
Comment (johnthecat) [inline src/renderer/shared/ui-kit/Inputs/common/types.ts]: Do we really need to move this outside of input props? Overall reading complexity outweighs benefits of shared configuration IMO

File: src/renderer/shared/ui-kit/Inputs/Input/Input.tsx [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+    const prefixRef = useRef<HTMLDivElement>(null);
+    const suffixRef = useRef<HTMLDivElement>(null);
+
+    const [paddingLeft, setPaddingLeft] = useState(DEFAULT_HORIZONTAL_PADDING);
+    const [paddingRight, setPaddingRight] = useState(DEFAULT_HORIZONTAL_PADDING);
+
+    useLayoutEffect(() => {
+      if (!prefixElement || !prefixRef.current) return;
+
+      setPaddingLeft(EXTENDED_HORIZONTAL_PADDING + prefixRef.current.getBoundingClientRect().width);
```
Comment (johnthecat) [inline src/renderer/shared/ui-kit/Inputs/Input/Input.tsx]: you can set ref to top level div element (also, it should be label) and remove all this complex code, made only to support combobox specific api.
Second option - keep ref for `input` element, but wrap `Input` into additional `div` inside `Combobox`

File: src/renderer/shared/ui-kit/Inputs/SearchInput/SearchInput.tsx [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import { forwardRef } from 'react';
+
+import { Icon, IconButton } from '@/shared/ui';
+import { Input, type InputProps } from '../Input/Input';
+
+type Props = Omit<InputProps, 'label' | 'prefixElement' | 'suffixElement'>;
+
+export const SearchInput = forwardRef<HTMLInputElement, Props>((props, ref) => {
+  return (
+    <Input
```
Comment (johnthecat) [inline src/renderer/shared/ui-kit/Inputs/SearchInput/SearchInput.tsx]: It should have `type="search"`

File: src/renderer/shared/ui-kit/Inputs/SearchInput/SearchInput.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+export default {
+  title: 'Design System/kit/SearchInput',
+  component: SearchInput,
+  parameters: { actions: { argTypesRegex: '^on.*' } },
+} as Meta<typeof SearchInput>;
+
+const Template: StoryFn<typeof SearchInput> = (args) => <SearchInput {...args} />;
+
+export const Primary = Template.bind({});
```
Comment (johnthecat) [inline src/renderer/shared/ui-kit/Inputs/SearchInput/SearchInput.stories.tsx]: Use new story format, please. Also, your stories are not interactive

File: src/renderer/shared/ui-kit/Inputs/TextArea/TextArea.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+  component: TextArea,
+  parameters: { actions: { argTypesRegex: '^on.*' } },
+} as Meta<typeof TextArea>;
+
+const Template: StoryFn<typeof TextArea> = (args) => <TextArea {...args} />;
+
+const LONG_TEXT =
+  'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Culpa doloribus iusto possimus praesentium ratione temporibus. Aperiam autem cumque esse eum fugit laborum quas! Architecto at, cupiditate dignissimos eveniet sunt voluptatibus.';
+
+export const Primary = Template.bind({});
```
Comment (johnthecat) [inline src/renderer/shared/ui-kit/Inputs/TextArea/TextArea.stories.tsx]: Use new story format

File: src/renderer/shared/ui-kit/Inputs/TextArea/TextArea.test.tsx
Comment (johnthecat) [inline src/renderer/shared/ui-kit/Inputs/TextArea/TextArea.test.tsx:1]: Move this test into storybook play function

File: src/renderer/shared/ui-kit/Inputs/TextArea/TextArea.tsx [hunk 11]
```diff
@@ -0,0 +1,29 @@
+import { type ComponentPropsWithoutRef, forwardRef } from 'react';
+
+import { cnTw } from '@/shared/lib/utils';
+import { type HTMLTextAreaProps } from '../common/types';
+
+interface Props extends Pick<ComponentPropsWithoutRef<'textarea'>, HTMLTextAreaProps> {
```
Comment (johnthecat) [inline src/renderer/shared/ui-kit/Inputs/TextArea/TextArea.tsx]: interface?

File: src/renderer/shared/ui-kit/Inputs/TextArea/TextArea.tsx [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+  invalid?: boolean;
+  onChange?: (value: string) => void;
```
Comment (johnthecat) [inline src/renderer/shared/ui-kit/Inputs/TextArea/TextArea.tsx]: Why onChange callback is optional? Component should be fully controllable IMO

File: src/renderer/features/balances/components/AmountInput.tsx [hunk 13]
```diff
@@ -196,3 +197,71 @@ export const AmountInput = ({
     />
   );
 };
+
+const DEFAULT_LEFT_PADDING = 11;
+const EXTENDED_LEFT_PADDING = 19;
+
+type InputProps = {
```
Comment (johnthecat) [inline src/renderer/features/balances/components/AmountInput.tsx]: Only `name | label | value | onChange` are used, maybe you should simplify this component?

File: src/renderer/features/balances/components/AmountInput.tsx (hunk 3 above)
Comment (tuul-wq) [reply src/renderer/features/balances/components/AmountInput.tsx]: Left 2 components in a single file, `AmountInput` requires refactoring anyway, did't want to do much while updating  `Inputs`

**Pull request merged**